contracts/StrategyVersionRegistry.vy

```diff
@@ -0,0 +1,90 @@
+# @version 0.2.12
+governance: public(address)
+pendingGovernance: public(address)
+
+event NewGovernance:
+    governance: address
+
+event NewPendingGovernance:
+    governance: address
+
+event StrategyRegistered:
+    strategy: address
+    name: String[64]
+    apiVersion: String[8]
+
+event Cloned:
+    clone: address
+
+interface Strategy:
+    def name() -> String[64]: view
+    def apiVersion() -> String[8]: view
+    def initialize(params: Bytes[256]): nonpayable
+    def isOriginal() -> bool: nonpayable
+
+strategyVersions: public(HashMap[String[73], address])
+
+@external
+def __init__():
+    self.governance = msg.sender
+
+@pure
+@internal
+def _computeKey(name: String[64], version: String[8]) -> String[73]:
+    return concat(name, "@", version)
+
+@external
+def setGovernance(governance: address):
+    """
+    @notice Starts the 1st phase of the governance transfer.
+    @dev Throws if the caller is not current governance.
+    @param governance The next governance address
+    """
+    assert msg.sender == self.governance  # dev: unauthorized
+    log NewPendingGovernance(msg.sender)
+    self.pendingGovernance = governance
+
+
+@external
+def acceptGovernance():
+    """
+    @notice Completes the 2nd phase of the governance transfer.
+    @dev
+        Throws if the caller is not the pending caller.
+        Emits a `NewGovernance` event.
+    """
+    assert msg.sender == self.pendingGovernance  # dev: unauthorized
+    self.governance = msg.sender
+    log NewGovernance(msg.sender)
+
+@external
+def addNewRelease(strategy :address, name: String[64] = ""):
+    """
+    @notice add a new version of a strategy
+    """
+    assert msg.sender == self.governance  # dev: unauthorized
+    assert Strategy(strategy).isOriginal() # dev: not original 
+    strategyName: String[64] = name 
+    if name == "":
+        strategyName = Strategy(strategy).name()
+        assert strategyName != ""
+    apiVersion: String[8] = Strategy(strategy).apiVersion()
+    key: String[73] = self._computeKey(strategyName, apiVersion)
+    self.strategyVersions[key] = strategy
```

Comment:
does it make sense to check if it is already set? 

thinking about issues that may come from overwriting a strategy@version with a different deployment

Reply:
It's expected to be overwritten as new version of a strategy are released.

Reply:
oh sure, I misinterpreted apiVersion for strategyVersion when writing this

still, any issues that may come from releasing a new version for the same StrategyName@ApiVersion pair? 

like getting all the existing clones from events and having 2 different deployments for same pair?

Reply:
Sorry didn't got it

Reply:
trying to think about potential issues that may arise from the following process
I release a strategy for yvDAI 0.4.2 
I add it to the registry StrategyA@0.4.2
Then I find a bug in that strategy and redeploy
I need to add it to the registry, the key will still be StrategyA@0.4.2 and it will override previous strategy

might this cause any issue ? 
cc @salazarguille

Reply:
That's the point is you keep updating the strategy address.